src/requestCertificate.jsx

```diff
@@ -121,24 +190,170 @@ const KeyFileRow = ({ onValueChanged, dialogValues }) => {
+    }
+
+    onValueChanged(key, value) {
+        const stateDelta = { [key]: value };
+
+        if (key === "dnsName") {
+            value = value.replace(/\s/g, ',');
+            while (value.includes(",,"))
+                value = value.replace(",,", ',');
+            stateDelta.dnsName = value;
+        }
+
+        this.setState(stateDelta);
+    }
+
+    checkParameters() {
+        return null;
+    }
+
+    onAddError(errorName, errorMessage) {
+        this.setState({ errorName, errorMessage });
+    }
+
+    onResubmit() {
+        const { cas, certPath } = this.props;
+        const casKeys = Object.keys(cas);
+        let caPath;
+        casKeys.forEach(key => {
+            if (cas[key].nickname.v === this.state.ca)
+                caPath = key;
+        });
+
+        let subjectName = this.state.subjectName;
+        if (subjectName && !subjectName.includes("="))
+            subjectName = "CN=" + subjectName;
+        let dnsNames;
+        if (this.state.dnsName) {
+            dnsNames = this.state.dnsName.split(',');
+            dnsNames = dnsNames.filter(Boolean); // Removes empty string entries
+        }
+
+        const parameter = { ca: cockpit.variant("s", caPath) };
+        parameter["template-subject"] = subjectName
+            ? cockpit.variant("s", subjectName)
+            : cockpit.variant("s", "");
+        parameter["template-principal"] = this.state.principalName
+            ? cockpit.variant("as", [this.state.principalName])
+            : cockpit.variant("as", [""]);
+        parameter["template-hostname"] = dnsNames
+            ? cockpit.variant("as", dnsNames)
+            : cockpit.variant("as", [""]);
+
+        modifyRequest(certPath, parameter)
+                .then(() => resubmitRequest(certPath))
+                .then(() => this.props.onClose())
+                .catch(error => this.onAddError(error.name, error.message));
+    }
+
+    render() {
+        const { onClose } = this.props;
+        const cas = Object.values(this.props.cas);
+
+        const body = (
+            <form className="ct-form">
+                {this.state.storage === "nssdb" && <>
+                    <label className="control-label" htmlFor="database-path-row">
+                        {_("Database path")}
+                    </label>
+                    <samp id="database-path-row">{ this.props.cert["cert-database"].v }</samp>
+                    <label className="control-label" htmlFor="storage-row">
+                        {_("Nickname")}
+                    </label>
+                    <samp id="nickname-row">{ this.state.nickname }</samp>
+                    <hr />
+                </>}
+
+                {this.state.storage === "file" && <>
+                    <label className="control-label" htmlFor="storage-row">
+                        {_("Certificate file")}
+                    </label>
+                    <samp id="cert-file-row">{ this.state.certFile }</samp>
```

Comment:
Little bit on the fence about using `<samp>` here.
According to the [MDN documentation on this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/samp) it _is used to enclose inline text which represents sample (or quoted) output from a computer program._, and I don't think that's quite the case here.
I think it would work better as a span for the path, similar to what we do on the services page.